### PR TITLE
M5515 Run molgenis-api-test tests against locally deployed war during integration-test fase

### PR DESCRIPTION
**Pull request by fdlk**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [nvt] User documentation updated
- [nvt] (If you have changed REST API interface) view-swagger.ftl updated
- [nvt] Test plan template updated
- [x] Clean commits

### CODE REVIEW

**Comment by dennishendriksen** [comment]
@fdlk what changes are required in order to merge with the master?

**Comment by fdlk** [comment]
@dennishendriksen we've a story on the board for this but basically a review + process comments

File: molgenis-api-tests/src/test/resources/conf/elasticsearch/elasticsearch.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# ====== MOLGENIS Elasticsearch Configuration ======
```
Comment (bartcharbon) [inline molgenis-api-tests/src/test/resources/conf/elasticsearch/elasticsearch.yml]: put somewhere central and copy to modules that need it

File: molgenis-api-tests/pom.xml [hunk 2]
```diff
@@ -12,7 +12,241 @@
...
+                                        <portName>elasticsearch.tcp.port</portName>
+                                        <portName>elasticsearch.http.port</portName>
+                                    </portNames>
+                                    <randomPort>true</randomPort>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <!-- start and stop elasticsearch instance -->
+                    <plugin>
```
Comment (bartcharbon) [inline molgenis-api-tests/pom.xml]: share this via the pom of the parent project as plugin configuration

File: molgenis-api-tests/pom.xml [hunk 3]
```diff
@@ -12,7 +12,241 @@
...
+                <!-- override the cluster name for the test run -->
+                <maven.build.timestamp.format>yyMMddHHmmssSSS</maven.build.timestamp.format>
+                <elasticsearch.cluster.name>apitest${maven.build.timestamp}</elasticsearch.cluster.name>
+                <elasticsearch.transport.addresses>localhost:${elasticsearch.tcp.port}
+                </elasticsearch.transport.addresses>
+            </properties>
+            <build>
+                <plugins>
+                    <!-- Reserve tcp and http ports for elasticsearch -->
+                    <plugin>
```
Comment (bartcharbon) [inline molgenis-api-tests/pom.xml]: share this via the pom of the parent project as plugin configuration

File: molgenis-api-tests/pom.xml [hunk 4]
```diff
@@ -12,7 +12,241 @@
...
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <!-- override the it db name and create new database for the test run -->
+                <maven.build.timestamp.format>yyMMddHHmmssSSS</maven.build.timestamp.format>
+                <it_db_name>molgenis_apitest_${maven.build.timestamp}</it_db_name>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
```
Comment (bartcharbon) [inline molgenis-api-tests/pom.xml]: share this via the pom of the parent project as plugin configuration

File: molgenis-api-tests/src/test/resources/home/molgenis-server.properties [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#environment=development
+user.password=admin
+admin.password=admin
+db_driver=org.postgresql.Driver
+db_uri=jdbc\:postgresql\://localhost/${it_db_name}?reWriteBatchedInserts\=true&autosave\=CONSERVATIVE
+db_user=${it_db_user}
+db_password=${it_db_password}
+mail.host=localhost
+mail.port=10025
```
Comment (bartcharbon) [inline molgenis-api-tests/src/test/resources/home/molgenis-server.properties]: make this dynamic

**Comment by fdlk** [comment]
Talked with @ConnorStroomberg , must also check if the data import tests still work if you specify the test to run manually.

**Comment by fdlk** [comment]
I've tested the parallel execution on my local machine by concurrently running `mvn clean verify` on the molgenis-api-tests module in two separate checkouts of this branch. Both runs succeed.

**Comment by fdlk** [comment]
And I've run the test with -Dtest= to explicitly point at the public dataset importer and verified that that overrides the exclusions in the failsafe plugin onfiguration.

**Pull request merged**